libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -1472,6 +1472,15 @@ bool AP_Logger::log_while_disarmed(void) const
         return true;
     }
 
+    static bool logged_armed = false; 
+    if (logged_armed) {
+        if (_params.log_disarmed == 3) { 
```

Comment:
we could probably simplify the logic with a simple 
if (_params.log_disarmed == 3 && arm_change_ms != 0) { return true;}
arm_change_ms != 0 means we arm once